Extension/src/LanguageServer/clientCollection.ts

```diff
@@ -154,14 +157,15 @@ export class ClientCollection {
                     // Transfer ownership of the client's documents to another client.
                     // (this includes calling textDocument/didOpen on the new client so that the server knows it's open too)
                     client.TrackedDocuments.forEach(document => this.transferOwnership(document, client));
-                    client.TrackedDocuments.clear();
 
                     if (this.activeClient === client && this.activeDocument) {
                         // Need to make a different client the active client.
                         this.activeClient = this.getClientFor(this.activeDocument.uri);
                         this.activeClient.activeDocumentChanged(this.activeDocument);
-                        // may not need this, the navigation UI should not have changed.
-                        // this.activeClient.selectionChanged(Range.create(vscode.window.activeTextEditor.selection.start, vscode.window.activeTextEditor.selection.end);
+                    }
+
+                    if (this.defaultClient === client) {
+                        this.defaultClient = this.getClientFor(folder.uri);
```

Comment:
Still working on this part.  Need to handle what happens if the user removes all folders and there isn't an existing client to make the default client.